osquery/tables/system/windows/logon_sessions.cpp

```diff
@@ -0,0 +1,195 @@
+      GetProcAddress(module, "LsaFreeReturnBuffer"));
+
+  if (LsaEnumerateLogonSessions && LsaGetLogonSessionData &&
+      LsaFreeReturnBuffer) {
+    PLUID sessions = NULL;
+    ULONG sessionCount = 0;
+    NTSTATUS status = LsaEnumerateLogonSessions(&sessionCount, &sessions);
+
+    for (size_t i = 0; i < sessionCount; i++) {
+      PSECURITY_LOGON_SESSION_DATA sessionData = NULL;
+      NTSTATUS status = LsaGetLogonSessionData(&sessions[i], &sessionData);
+
+      // I need to go back and convert LARGE_INTEGER values to Unix Timestamp
+      if (status == 0) {
+        Row r;
+        r["logon_id"] = INTEGER(sessionData->LogonId.LowPart);
+        r["user"] = wstringToString(sessionData->UserName.Buffer);
+        r["logon_domain"] = wstringToString(sessionData->LogonDomain.Buffer);
+        r["authentication_package"] =
+            wstringToString(sessionData->AuthenticationPackage.Buffer);
+        r["logon_type"] = GetLogonType(sessionData->LogonType);
+        r["session_id"] = INTEGER(sessionData->Session);
+        LPTSTR sid;
+        ConvertSidToStringSid(sessionData->Sid, &sid);
+        r["logon_sid"] = sid;
+        r["logon_time"] = BIGINT(filetimeToUnixtime(sessionData->LogonTime));
+        r["logon_server"] = wstringToString(sessionData->LogonServer.Buffer);
+        r["dns_domain_name"] =
+            wstringToString(sessionData->DnsDomainName.Buffer);
+        //r["upn"] = wstringToString(sessionData->Upn.Buffer);
+        r["upn"] = INTEGER(sessionData->Upn.Length);
+        /*
+				r["user_flags"] = INTEGER(sessionData->UserFlags); 
+				r["last_successful_logon"] = BIGINT(
+            filetimeToUnixtime(sessionData->LastLogonInfo.LastSuccessfulLogon));
+        r["last_failed_logon"] = BIGINT(
+            filetimeToUnixtime(sessionData->LastLogonInfo.LastFailedLogon));
+        r["failed_attempt_count_since_last_successful_logon"] =
+            INTEGER(sessionData->LastLogonInfo
+                        .FailedAttemptCountSinceLastSuccessfulLogon);
+        */
+				r["logon_script"] = wstringToString(sessionData->LogonScript.Buffer);
+        r["profile_path"] = wstringToString(sessionData->ProfilePath.Buffer);
+        r["home_directory"] =
+            wstringToString(sessionData->HomeDirectory.Buffer);
+        r["home_directory_drive"] =
+            wstringToString(sessionData->HomeDirectoryDrive.Buffer);
+        /*
+				r["logoff_time"] = BIGINT(filetimeToUnixtime(sessionData->LogoffTime));
+        r["kick_off_time"] =
+            BIGINT(filetimeToUnixtime(sessionData->KickOffTime));
+        r["password_last_set"] =
+            BIGINT(filetimeToUnixtime(sessionData->PasswordLastSet));
+        r["password_can_change"] =
+            BIGINT(filetimeToUnixtime(sessionData->PasswordCanChange));
+        r["password_must_change"] =
+            BIGINT(filetimeToUnixtime(sessionData->PasswordMustChange));
+        */
+				results.push_back(r);
```

Comment:
Indentation is off.

Reply:
I think you may have some tabs in here instead of spaces. Convert everything to spaces.